### ISSUE TITLE
Unify trailing slashes in generated java constants

### ISSUE THREAD

**Issue by hiddenalpha** [swisspush/apikana#30]
It seems we've no convention when generated java constants have trailing slashes and when they don't. Even old and new Path classes are inconsistent. That introduced lot extra work for our services upgrading the APIs and still gets us in trouble where those inconsistencies result in unexpected paths and most time aren't discovered until malformed services reach the productive stage.

My suggestion:
- ALWAYS use trailing slashes for collections
- NEVER use trailing slashes for resources
- NEVER use leading slashes

Originally we expected old and new Paths are fully compatible. But in table below we collected some examples and now see some (for us evil) differences. Our expectation was that each pair for _leading_ and _trailing_ will be same for old and new Paths and there's also some consistency within same version. But our expectation seems wrong.

Path | Type | leading old | leading new |     | trailing old | trailing new |
---- | ---- |:-----------:|:-----------:| --- |:------------:|:------------:|
mediamessage/v1/config/ | Collection | no | no |  | yes | no
mediamessage/v1/config/suppressmessageflow | Resource | no | no |  | no | no
suppressmessageflow | Resource | no | yes |  | no | no
mediamessage/v1/online/{mediaChannel}/ | Collection | no | no |  | yes | no
mediamessage/v1/online/{mediaChannel}/{stopId}/ | Collection | no | no |  | yes | no
mediamessage/v1/online/{mediaChannel}/{stopId}/{stopIndex} | Resource | no | no |  | no | no
{stopIndex} | Resource | no | yes |  | no | no
stopIndex | Resource | no | no |  | no | no
{stopId} | Collection | no | yes |  | no | no
stopId | Collection | no | no |  | no | no
{stopId}/{stopIndex} | Resource | no | yes |  | no | no
{mediaChannel}/{stopId}/{stopIndex} | Resource | no | yes |  | no | no

**Comment by hiddenalpha** [comment]
It seems we've a conflict with these two conditions:

- ALWAYS use trailing slashes for collections
- NEVER use trailing slashes for resources

Combined with following endpoints:
```
paths:
  "/example/v1/pets":
    post: // Create a new pet
  "/example/v1/pets/{petId}":
    get: // Get a specific pet
```
Now `/example/v1/pets` badly looks like both: A resource (where we POST new content to) and a collection (where we can select a specific entry by _petId_.

Now the question is: What actually should resolve `ExampleApiPaths.pets.PATH` to? Which variant we expect depends on the context where this constant actually gets used:
- /example/v1/pets  ->  In case we want to use path for a POST
- /example/v1/pets/  ->  In case we want to append _petId_ to get a specific resource.

But because this is only once constant, we can only provide one of them, breaking at least one of the rules named above.

**Comment by hiddenalpha** [comment]
We'll adapt the requirement as follows:
- _~ALWAYS use trailing slashes for collections~_
- **NEVER use trailing slashes** _~for resources~_
- **NEVER use leading slashes**

**Comment by hiddenalpha** [comment]
Won't fix this in _2nd generation_ due to breaking changes.

Instead implemented _3rd generation_ to address it. See: #32.
